changelog.d/19712_propagate_tracing_span_context_stream_sink_request_building.fix.md

```diff
@@ -0,0 +1 @@
+Vector logs and internal metrics emitted while in the context of stream sink request building, now correctly contain the component's automatic tags (`component_kind`, `component_type`, `component_id`).
```

Comment:
I don't think a user will know what "stream sink request building" is or how that affects them. I would say something like:

The following metrics now correctly have the `component_kind`, `component_type`, and `component_id` tags:

- X
- Y

For sinks:

- A
- B
- C